### ISSUE TITLE
 Unable to load rule assembly 

### ISSUE THREAD

**Issue by balubchand** [dotnet/roslyn#4073]
I have created a live code analyser using sdk template in Visual studio 2015 RC(https://msdn.microsoft.com/en-us/magazine/dn879356.aspx).  I was able to create nuget package out of it and install the same into class library(Created this using  VS 2015 RTM). Live code analyser is working like a charm.

I tried to create a ruleset for the same and tried to integrate the same in the build of the project by changing the property of the project.
Problem started from there. I am getting the message below
1>  Running Code Analysis...
1>MSBUILD : error : CA0053 : Unable to load rule assembly 'c:\myprojects\classlibrary1\packages\eyanalyser.1.0.0.7\tools\analyzers\eyanalyser.dll': Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
1>MSBUILD : warning : CA0064 : No analysis was performed because the specified rule set could not be loaded or did not contain any managed code analysis rules.

The ruleset  is 
```<RuleSet Name="Project WizBang Rules" ToolsVersion="1.1">
 <RuleHintPaths>
     <Path>C:\MyProjects\ClassLibrary1\packages\EYAnalyser.1.0.0.7\tools\analyzers</Path>
   </RuleHintPaths>
  <Rules AnalyzerId="EYAnalyser" RuleNamespace="EYAnalyser">
    <Rule Id="3.4 NameSpace" Action="Warning" />
    <Rule Id="4.2 Class" Action="Warning" />
    <Rule Id="4.6 Field Naming" Action="Warning" />
  </Rules>
</RuleSet>```

**Comment by balubchand** [comment]
Update: On further investigation
I tried to load the assembly in the nugget package using Assembly.Load(), although the load was successful. I could see the error in assembly DefinedTypes.LoaderExceptions, 

{"Could not load file or assembly 'Microsoft.CodeAnalysis.Workspaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.":"Microsoft.CodeAnalysis.Workspaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"}

{"Could not load file or assembly 'Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.":"Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"}	System.Exception {System.IO.FileNotFoundException}

**Comment by tmeschter** [comment]
@balubchand Diagnostic analyzers built on top of Roslyn are completely different from the old Code Analysis rules ("FxCop"). FxCop does binary analysis *after* the compiler has run and produced a binary, while Roslyn diagnostic analyzers run *inside* the compiler and examine the source code (as syntax trees and symbols).

You appear to be telling FxCop to load your diagnostic analyzer by using the `<RuleHintPaths>` element of the rule set file. This will not work, as FxCop has no knowledge of Roslyn.

Instead, you should add your analyzer to the project using the "Add Analyzer..." menu item in VS. It's near the "Add Reference..." menu item. Once that is done, every time you build your project the C# or VB compiler will run your analyzer.

Roslyn diagnostic analyzers *do* still use the same .ruleset files used by FxCop to turn individual diagnostics on and off and change their severity.

**Comment by tmeschter** [comment]
@balubchand Sorry, I got this issue confused with the other one you filed. If you've already installed your analyzer as a NuGet package, I think you just need to remove the `<RuleHintPaths>` item from the .ruleset file so that FxCop stops trying to load the analyzer.

**Comment by balubchand** [comment]
@tmeschter when I remove the <RuleHintPaths> I am getting an error as below. Let me rephrase live code analysis is working fine, but I am  get this warning when I click on analyse or configure code analysis on build.
When I open the active ruleset also I am not able to view my custom rule.

Warning	CA0064	CA0064 : No analysis was performed because the specified rule set could not be loaded or did not contain any managed code analysis rules.

**Comment by mavasani** [comment]
@balubchand Do you see the same issue on VS2015 RTM? I think you shared your EYCodeRuleAnalyzer with me and Deepak, and are able to use the analyzer + ruleset in your sample Test project.

Please re-open if you are seeing the issue on VS2015 RTM.

**Comment by mavasani** [comment]
Verified by the customer.
